src/parsers/minecraft/coord/coord_base.ts

```diff
@@ -0,0 +1,129 @@
+            }
+            switch (reader.peek()) {
+                case "~":
+                    if (hasLocal) {
+                        return helper.fail(
+                            MIXED.create(reader.cursor, reader.cursor + 1)
+                        );
+                    }
+                    hasWorld = true;
+                    reader.skip();
+                    if (!helper.merge(this.parseNumber(reader))) {
+                        return helper.fail();
+                    }
+                    break;
+                case "^":
+                    if (hasWorld) {
+                        return helper.fail(
+                            MIXED.create(reader.cursor, reader.cursor + 1)
+                        );
+                    }
+                    if (!this.rules.local) {
+                        return helper.fail(
+                            NO_LOCAL.create(reader.cursor, reader.cursor + 1)
+                        );
+                    }
+                    hasLocal = true;
+                    reader.skip();
+                    if (!helper.merge(this.parseNumber(reader))) {
+                        return helper.fail();
+                    }
+                    break;
+                default:
+                    if (hasLocal) {
+                        return helper.fail(
+                            MIXED.create(reader.cursor, reader.cursor + 1)
+                        );
+                    }
+                    if (!reader.canRead()) {
+                        helper.addSuggestions(
+                            {
+                                start: reader.cursor,
+                                text: "~"
+                            },
+                            {
+                                start: reader.cursor,
+                                text: "^"
+                            }
+                        );
+                    }
+                    hasWorld = true;
+                    if (!helper.merge(this.parseNumber(reader, false))) {
+                        return helper.fail();
+                    }
+            }
+            if (
+                (!reader.canRead() || !helper.merge(reader.expect(" "))) &&
+                i < this.rules.count - 1
+            ) {
+                return helper.fail(
+                    INCOMPLETE.create(
+                        start,
+                        reader.cursor,
+                        i + 1,
+                        this.rules.count
+                    )
+                );
+            }
+        }
+        return helper.succeed();
+    }
+
+    private parseNumber(
+        reader: StringReader,
+        allowBlank: boolean = true
+    ): ReturnedInfo<number> {
+        if ((!reader.canRead() || reader.peek().match(/\s/)) && allowBlank) {
+            return new ReturnHelper().succeed(0);
+        }
+        return this.rules.float ? reader.readFloat() : reader.readInt();
+    }
+}
+
+export interface CoordRules {
```

Comment:
Could you declare this at the top of the file please. It's first use is their, so it might be more clear.